src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -594,7 +600,25 @@ public RowMap getRow() throws Exception {
 				case QUERY:
 					QueryEventData qe = event.queryData();
 					String sql = qe.getSql();
-					if (BinlogConnectorEvent.BEGIN.equals(sql)) {
+					if (sql.startsWith(BinlogConnectorEvent.XASTART) || sql.startsWith(BinlogConnectorEvent.XABEGIN)) {
+						String xid = getXAid(sql);
+						LOGGER.info(BinlogConnectorEvent.XASTART + " rowBufferHashMapXA.size(): " + rowBufferHashMapXA.size() + " xid " + xid);
+						RowMapBuffer rowMapBufferXA = getTransactionRows(event);
+						rowBufferHashMapXA.put(xid, rowMapBufferXA);
+					} else if (sql.startsWith(BinlogConnectorEvent.XACOMMIT)) {
+						String xid = getXAid(sql);
+						rowBuffer = rowBufferHashMapXA.remove(xid);
+						updateStatus(rowBuffer);
+						LOGGER.info(BinlogConnectorEvent.XACOMMIT + " rowBufferHashMapXA.size(): " + rowBufferHashMapXA.size() + " xid " + xid);
+					} else if (sql.startsWith(BinlogConnectorEvent.XAROLLBACK)) {
+						String xid = getXAid(sql);
+						RowMapBuffer rowMapBuffer = rowBufferHashMapXA.remove(xid);
+						while (!rowMapBuffer.isEmpty()) {
+							RowMap r = rowMapBuffer.removeFirst();
+							LOGGER.warn("XA ROLLBACK RowMap Table Name:" + r.getTable() + ", xid: " + xid + ", Json: " + r.toJSON());
```

Comment:
it doesn't seem like a great idea to do all these warnings.

Reply:
Yes! This log is only belong to my system! I forget remove this part(log warnings).